### PR TITLE
Fixing build errors & formatting

### PR DESCRIPTION
**Pull request by nate-double-u**
I think this sorts out the broken links in the main PR. I've also gone through and reformatted some tables and corrected some typos I saw.

I'm not sure exactly where all the links are supposed to go, so in some cases I've just removed them, a cue to find where they should go.

### CODE REVIEW

File: content/docs/2.15/deploy.md [hunk 1]
```diff
@@ -4,9 +4,9 @@ title = "Deploying KEDA"
 
 We provide a few approaches to deploy KEDA runtime in your Kubernetes clusters:
 
-- [Helm charts](#helm)
-- [Operator Hub](#operatorhub)
-- [YAML declarations](#yaml)
```
Comment (nate-double-u) [inline content/docs/2.15/deploy.md:9]: I'm not sure where these should link, so i've just removed the links

File: content/docs/2.15/operate/_index.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ We provide guidance and requirements around various areas to operate KEDA:
 
 - Admission Webhooks ([link](./admission-webhooks))
 - Cluster ([link](./cluster))
-- Kubernetes Events ([link](./events))
```
Comment (nate-double-u) [inline content/docs/2.15/operate/_index.md:11]: Same here, i'm not sure where this should go, so I've removed the link.

File: content/docs/2.15/reference/_index.md [hunk 3]
```diff
@@ -6,8 +6,8 @@ weight = 2
 Reference information for the KEDA autoscaler.
 
 - [Scaled Object specification](./scaledobject-spec)
-- [ScaledJob specification](../concepts/scaling-jobs/#scaledjob-spec)
+- [ScaledJob specification](./scaledjob-spec.md)
```
Comment (nate-double-u) [inline content/docs/2.15/reference/_index.md:9]: I think this is correct, but it should be double checked

File: content/docs/2.15/deploy.md (hunk 1 above)
Comment (dwelsch-esi) [reply content/docs/2.15/deploy.md:9]: They're explicitly tagged anchors internal to this file. For example, line 15:
```
## Deploying with Helm {#helm}
```

They work in the existing documentation. If they don't break anything I'd leave them in.

File: content/docs/2.15/operate/_index.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ We provide guidance and requirements around various areas to operate KEDA:
...
+- Kubernetes Events
```
Comment (dwelsch-esi) [inline content/docs/2.15/operate/_index.md]:
```suggestion
- Kubernetes Events ([link](../reference/events))
```

File: content/docs/2.15/reference/_index.md (hunk 3 above)
Comment (dwelsch-esi) [reply content/docs/2.15/reference/_index.md:9]: Yup, that's correct.

File: content/docs/2.15/reference/_index.md [hunk 5]
```diff
@@ -6,8 +6,8 @@ weight = 2
...
 - [Events]
 - [Firewall requirements]
 - [FAQ](./faq.md)
-- [Glossary](./glossary.md)
+- [Glossary](./glossary.md)
```
Comment (dwelsch-esi) [inline content/docs/2.15/reference/_index.md:13]: Brrw?

**Review by dwelsch-esi — COMMENTED** [review]
See comments and suggestions regarding removed links. Otherwise LGTM.

File: content/docs/2.15/deploy.md (hunk 1 above)
Comment (dwelsch-esi) [reply content/docs/2.15/deploy.md:9]: Update: As you point out, they might be what's causing the reference problem, so let's remove them and see what happens.

File: content/docs/2.15/operate/_index.md (hunk 2 above)
Comment (dwelsch-esi) [reply content/docs/2.15/operate/_index.md:11]: This one I'm sure of. I moved the events to the reference directory.

File: content/docs/2.15/reference/_index.md (hunk 5 above)
Comment (dwelsch-esi) [reply content/docs/2.15/reference/_index.md:13]: Stray space? Accepting change; won't affect the doc.

**Pull request merged**